bin/send_email.py

```diff
@@ -37,12 +37,12 @@ def get_participant_id(syn: synapseclient.Synapse, submission_id: str) -> List[s
 
 
 def get_score_dict(score):
-    strs = [""]
+    strings = [""]
     for key in score.keys():
-        str = f"{key} : {score[key][0]}" + "\n"
-        strs.append(str)
+        string = f"{key} : {score[key][0]}" + "\n"
+        strings.append(string)
 
-    return strs
+    return strings
 
 
 def email_template(
```

Comment:
For the line:
```
f"\nView all your scores here: https://www.synapse.org/#!Synapse:{view_id}/tables/",
```

Typically, I query results directly from the evaluation queue on the live site, allowing participants to view their scores (e.g., SELECT * FROM evaluation_9615379 WHERE status == 'ACCEPTED' AND score_status == 'SCORED'). This approach prevents the need to share the view table with participants, in order to keep other participants' results confidential, especially during the validation phase of certain challenges.

With the new wiki structure for the fitting challenge portal, which includes different tabs for individual tasks. I don't think a fixed wiki link is feasible, either. So, I suggest redirecting participants to the live site for an optimal solution?

Reply:
I think if participants could see the view they would only be able to see their own submissions (this is the approach developed for the PEGS Challenge), though I could be mistaken. 

Where precisely would you want participants to be directed on the live site? And are you opposed to scores being displayed in the email notification?

Reply:
> I think if participants could see the view they would only be able to see their own submissions (this is the approach developed for the PEGS Challenge), though I could be mistaken.

I am not sure about that, either. I think @thomasyu888 and @vpchung have more experience with permissions of submission views.  Based on my experience, when I logged in with my testing account as a participant who only had read access to the table, I was able to see the results of others in the table.

<img width="914" alt="Screen Shot 2024-02-22 at 12 07 16 PM" src="https://github.com/Sage-Bionetworks-Workflows/nf-synapse-challenge/assets/73901500/1399a972-9936-4c4b-ad6a-faf05bf8b168">

> Where precisely would you want participants to be directed on the live site? And are you opposed to scores being displayed in the email notification?

I prefer to redirect to a "[Submissions](https://www.synapse.org/#!Synapse:syn52052735/wiki/626195)" or "[Results leaderboard](https://www.synapse.org/#!Synapse:syn52052735/wiki/626197)", but as I previously mentioned, the results from different evaluations will be on separate wiki pages.

To clarify, for this challenge, I would like to include scores in the emails sent to participants (and share view table with participants). I am considering a robust way to redirect their results in the email for other challenges that require hiding scores from other participants.

Reply:
That's odd considering I was unable to see anything in the submission view until you specifically gave me access to the evaluation queues themselves...

Reply:
Would you be able to test this out with your testing account?

Reply:
Oh, I see. I didn't realize that account had those exact permissions already. Thanks!

Reply:
I'm late to the party, but if it helps any, we do not currently link participants back to the leaderboard in the email; we only show the scores they have received (if it's allowed).  Otherwise, we'd notify them that their submission has been evaluated.

That being said, I can see the link as a useful feature moving forward.  I agree with Rong though, in that I prefer to send them to the leaderboard that we have curated for them, rather than directly to the Synapse table.